src/Microsoft.AspNet.BasicMiddleware/HttpMethodOverrideMiddleware.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http;
+
+namespace Microsoft.AspNet.BasicMiddleware
+{
+    public class HttpMethodOverrideMiddleware
+    {
+        private const string xHttpMethodOverride = "X-Http-Method-Override";
+        private readonly RequestDelegate _next;
+        private readonly string _inputName;
+
+        public HttpMethodOverrideMiddleware(RequestDelegate next, string inputName = null)
+        {
+            _next = next;
+            _inputName = inputName;
+
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (_inputName != null)
+            {
+                var form = await context.Request.ReadFormAsync();
+                var methodType = form[_inputName];
+                context.Request.Method = methodType;
+                await _next(context);
```

Comment:
make some changes to not duplicate the await and the assignment to Request.method